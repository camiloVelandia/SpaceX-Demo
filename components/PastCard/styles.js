import styled from "styled-components";

export const Section = styled.article`
  margin-bottom: 40px;
  border: 1px solid white;
  text-align: center;
  width: 100%;
  @media (min-width: 768px) {
  width: 48%;
  }
`;
export const Title = styled.h2`
  background-color:white;
  color:black;
  padding:30px 0;
  font-size:36px;
  text-transform:uppercase;
  font-weight:600;
  margin:0;
`;
export const Figure = styled.figure`
  max-width:228px;
  max-height:218px;
  height:100%;
  margin:0 auto;
  margin-top:30px;
  & img{
    height:100%;
    object-fit:cover;
  }
`;
export const Name = styled.h3`
  font-size:24px;
  margin-top:30px;
  margin-bottom:30px;
  text-transform:uppercase;
  padding:0 20px;
  margin:0;
`;
export const Text = styled.p`
padding:20px 20px;
text-align:justify;
padding-bottom:30px;
  margin:0;
`;
export const Info = styled.div`
  padding:0 20px;
  padding-bottom:20px;
  display:flex;
  justify-content:space-between;
`;
export const Bullet = styled.div`
  & p {
    margin: 0;
    text-align: left;
    font-size:12px;
    padding:0 10px;
  }
  @media (min-width: 768px) {
    & p {
    font-size:18px;
      margin: 0;
      text-align: left;
    }
  }
`;
