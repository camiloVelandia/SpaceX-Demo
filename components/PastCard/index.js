import React from 'react';
import {Section, Title, Figure,Name,Text,Info,Bullet} from './styles'

const PastCard = () => {
  return (
    <Section>
      <Title>recent missions</Title>
      <Figure>
        <img src={"/images/escudo.png"} />
      </Figure>
      <Name>name mission</Name>
      <Text>
        Lorem ipsum dolor sit amet consectetur adipiscing elit montes a
        pellentesque felis, at sed quam quis praesent tempus arcu sociis
        habitasse donec eros, lacinia elementum etiam per faucibus bibendum
        lacus venenatis leo nam. Nec aliquet eros phasellus hendrerit consequat
        magnis malesuada nascetur posuere at, accumsan vestibulum cum gravida
        erat eget congue imperdiet sagittis, lacinia taciti ultricies mi pretium
      </Text>
      <Info>
        <Bullet>
          <p>date:</p>
          <p>november 10</p>
        </Bullet>
        <Bullet>
          <p>date:</p>
          <p>november 10</p>
        </Bullet>
        <Bullet>
          <p>date:</p>
          <p>november 10</p>
        </Bullet>
        
      </Info>
    </Section>
  );
};

export default PastCard;