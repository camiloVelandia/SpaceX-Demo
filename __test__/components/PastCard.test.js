import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import PastCard from "../../components/PastCard";

describe("<PastCard/>", () => {
  const pastcard = mount(<PastCard />);
  test("render del componente pastcard", () => {
    expect(pastcard.length).toEqual(1);
  });
});

describe("pastcard snaaapshot", () => {
  test("Comprobar ui del pastcard", () => {
    const pastcard = create(<PastCard />);
    expect(pastcard.toJSON()).toMatchSnapshot();
  });
});
