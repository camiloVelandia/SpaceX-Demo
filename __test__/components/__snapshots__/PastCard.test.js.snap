// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pastcard snaaapshot Comprobar ui del pastcard 1`] = `
<article
  className="styles__Section-sc-1ltfs9r-0 cpXMDO"
>
  <h2
    className="styles__Title-sc-1ltfs9r-1 gNdGHw"
  >
    recent missions
  </h2>
  <figure
    className="styles__Figure-sc-1ltfs9r-2 hKcPoF"
  >
    <img
      src="/images/escudo.png"
    />
  </figure>
  <h3
    className="styles__Name-sc-1ltfs9r-3 gclgRl"
  />
  <p
    className="styles__Text-sc-1ltfs9r-4 gkoFmq"
  />
  <div
    className="styles__Info-sc-1ltfs9r-5 irsmnC"
  >
    <div
      className="styles__Bullet-sc-1ltfs9r-6 buZOHz"
    >
      <p>
        <span>
          date:
        </span>
      </p>
      <p>
        
      </p>
    </div>
    <div
      className="styles__Bullet-sc-1ltfs9r-6 buZOHz"
    >
      <p>
        <span>
          flight number:
        </span>
      </p>
      <p />
    </div>
    <div
      className="styles__Bullet-sc-1ltfs9r-6 buZOHz"
    >
      <p>
        <span>
          succes:
        </span>
      </p>
      <p />
    </div>
  </div>
</article>
`;
